apis/python/src/tiledbsoma/tiledb_object.py

```diff
@@ -73,12 +67,13 @@ def _ctx(self) -> tiledb.Ctx:
 
     @property
     def metadata(self) -> MetadataMapping:
-        """Metadata accessor"""
+        # This needs to be implemented as a @property because Python's ABCs
+        # require that abstract properties be implemented on the object itself,
+        # rather than being a field (i.e., creating self.whatever in __init__).
         return self._metadata
-        # Note: this seems trivial, like we could just have `metadata` as an attribute.
-        # However, we've found that since in `somacore` it's implemented as `@property`,
-        # to avoid a static-analysis failure we have to do the same here.
 
+    # TODO: This needs reconsidering, since it means we're deleting something
+    # out from underneath ourselves.
```

Comment:
should be a class method

Reply:
I've removed it entirely for now; when we restore it it should just be `tiledbsoma.delete` since it's not attached to a type.